src/braket/quantum_information/pauli_string.py

```diff
@@ -142,6 +147,138 @@ def eigenstate(self, signs: Optional[Union[str, List[int], Tuple[int, ...]]] = N
+
+        Args:
+            other (PauliString): The right multiplicand.
+            inplace (bool): If `True`, `self` is updated to hold the product.
+
+        Returns:
+            PauliString: The resultant circuit from right multiplying `self` with `other`.
+
+        Raises:
+            ValueError: If the lengths of the Pauli Strings being multiplied differ.
+        """
+        if self._qubit_count != other._qubit_count:
+            raise ValueError(
+                f"Input Pauli string must be of length ({self._qubit_count}), "
+                f"not {other._qubit_count}"
+            )
+        pauli_result = ""
+        phase_result = self._phase * other._phase
+        for i in range(self._qubit_count):
+            # Are either identity?
+            if i not in self._nontrivial and i not in other._nontrivial:
+                pauli_result += "I"
+            elif i not in self._nontrivial:
+                pauli_result += other._nontrivial[i]
+            elif i not in other._nontrivial:
+                pauli_result += self._nontrivial[i]
+            elif self._nontrivial[i] == other._nontrivial[i]:
+                pauli_result += "I"
+            else:
+                gate, phase = _PRODUCT_MAP[self._nontrivial[i]][other._nontrivial[i]]
+                pauli_result += gate
+                phase_result *= phase
+
+        # ignore complex global phase
+        if phase_result.real < 0 or phase_result.imag < 0:
+            pauli_result = "-" + pauli_result
+        out_pauli_string = PauliString(pauli_result)
+
+        if inplace:
+            self._phase = out_pauli_string._phase
+            self._qubit_count = out_pauli_string._qubit_count
+            self._nontrivial = out_pauli_string._nontrivial
+        return out_pauli_string
+
+    def __mul__(self, other: PauliString) -> PauliString:
+        """Right multiplication operator overload using `dot()`.
+
+        Returns the result of multiplying the current circuit by the argument on its right.
+
+        Args:
+            other (PauliString): The right multiplicand.
+
+        Returns:
+            PauliString: The resultant circuit from right multiplying `self` with `other`.
+
+        Raises:
+            ValueError: If the lengths of the Pauli Strings being multiplied differ.
+
+        See Also:
+            `braket.quantum_information.PauliString.dot()`
+        """
+        return self.dot(other)
+
+    def power(self, n: int, inplace: bool = False) -> PauliString:
+        """Composes circuit with itself n times.
+
+        Args:
+            n (int): The number of times to self-multiply. Must be greater than 0.
+            inplace (bool): Update `self` if `True`
+
+        Returns:
+            PauliString: The circuit from right multiplying `self` with itself `n` times.
+
+        Raises:
+            ValueError: If `n <= 0`.
+        """
+        if n <= 0:
+            raise ValueError("Must be raised to power strictly greater than 0")
```

Comment:
Was copying how `qiskit.quantum_info.Pauli.power` handles raising to non-positive integers, but your suggestions make sense. Inverses should be trivial since Pauli ops are involutory. Will commit.